sklearn/model_selection/_split.py

```diff
@@ -913,6 +915,238 @@ def get_n_splits(self, X, y, groups):
         return int(comb(len(np.unique(groups)), self.n_groups, exact=True))
 
 
+class _RepeatedSplits(with_metaclass(ABCMeta)):
+    """Repeated splits for K-Fold and Stratified K-Fold
+
+    Repeats splits for cross-validators n times.
+
+    Parameters
+    ----------
+    cv : object, KFold or StratifiedKFold
+        Instance of cross-validator.
+
+    n_repeats : int, default=5
+        Number of times cross-validator needs to be repeated.
+
+    random_state : None, int or RandomState, default=None
+        Random state to be used to generate random state for each
+        repetition.
+    """
+    def __init__(self, cv, n_repeats=5, random_state=None):
+        if not isinstance(cv, (KFold, StratifiedKFold)):
+            raise ValueError(
+                "cv must be an instance of KFold or StratifiedKFold.")
```

Comment:
I think only `KFold` and `StratifiedKFold` use `random_state`. That's why I added this check. Should I remove it? And also, is there a way to check if `cv` is an instance of cross-validator with randomized split functionality?